libraries/search/lib/objective/ObjectiveFunction.ts

```diff
@@ -24,18 +24,52 @@ import { SearchSubject } from "../SearchSubject";
  *
  * @author Mitchell Olsthoorn
  */
-export interface ObjectiveFunction<T extends Encoding> {
+export abstract class ObjectiveFunction<T extends Encoding> {
+  /**
+   * Indicates if the distance should be shallow or deep.
+   *
+   * Shallow distances only return 0 (covered) or 1 (uncovered).
+   * Deep distances return the actual distance.
+   *
+   * Shallow distances can be used to speed up the search process
+   * when an objective is already covered while still improving the solution.
+   */
+  protected shallowDistance: boolean;
+
+  constructor() {
```

Comment:
We can move the id to this base class since we use abstract class over interface now.
We can also implement getIdentifier here.